### DISCUSSION TITLE
2022.216.0+ unable to open on MacOS High Sierra 10.13.6

### DISCUSSION THREAD

**Discussion by Agrrox** [ppy/osu#17665] [Q&A] [answered]
### Type

Other

### Bug description

All builds starting with [2022.216.0](https://github.com/ppy/osu/releases/tag/2022.216.0) cannot be opened on High Sierra 10.13.6. 
But previously, there wasn't this problem. 
A lot of times this is an artificial limitation set somewhere inside the app bundle.
 
I'm unable to update my device. Is there a way to build/deploy osu!lazer for 10.13.6?

**EDIT:** I'm able to open the game by running the executable in `/osu!.app/Contents/MacOS/osu!` so this must be artificial limitation somewhere.

### Screenshots or videos

![Screen Shot 2022-04-05 at 18 31 12](https://user-images.githubusercontent.com/10901821/161803402-f55e5a7e-8f68-4604-a3f8-71451621c278.png)
![Screen Shot 2022-04-05 at 18 30 39](https://user-images.githubusercontent.com/10901821/161803415-0559d46b-07a3-4f03-86b1-38585e3239a3.png)

### Version

2022.216.0

### Logs

There are no logs.

**Comment by frenzibyte** [comment, marked as answer]
As mentioned in the message (albeit slightly off), osu!(lazer) can run on macOS 10.15 and onwards:

<img width="331" alt="CleanShot 2022-04-05 at 19 40 15@2x" src="https://user-images.githubusercontent.com/22781491/161803842-04077338-8935-4b01-a8ca-580c6ea067c7.png">

This is a .NET level limitation, as .NET 5 and .NET 6 only support macOS 10.15 at minimum.

**Comment by Agrrox** [comment]
@frenzibyte 
I'm able to open and successfully run the application by running the executable inside the app bundle `/osu!.app/Contents/MacOS/osu!` so this must be an artificial limitation or a bug.

```
Last login: Tue Apr  5 18:41:20 on ttys000
mac:~ macpro$ /Users/macpro/Downloads/osu\!.app/Contents/MacOS/osu\! ; exit;
2022-04-05 18:45:27.162 osu![19155:744847] NSDocumentController Info.plist warning: The values of CFBundleTypeRole entries must be 'Editor', 'Viewer', 'None', or 'Shell'.
2022-04-05 18:45:27.162 osu![19155:744847] NSDocumentController Info.plist warning: The values of CFBundleTypeRole entries must be 'Editor', 'Viewer', 'None', or 'Shell'.
2022-04-05 18:45:27.162 osu![19155:744847] NSDocumentController Info.plist warning: The values of CFBundleTypeRole entries must be 'Editor', 'Viewer', 'None', or 'Shell'.
2022-04-05 16:45:30.461 Info: Realm at path /Users/macpro/.local/share/osu/client.realm was found to contain Guid values in little-endian format and was automatically migrated to store them in big-endian format.
```
![Screen Shot 2022-04-05 at 18 48 11](https://user-images.githubusercontent.com/10901821/161806006-f07abfbc-e835-4dce-bccc-935a6a3ae59f.png)

**Comment by frenzibyte** [comment]
As I mentioned, it is not a limit we intentionally set, and there is nothing we have locally that sets such limit in the first place, so you will have to either continue using your current way of running it, or update your machine and call it a day.

**Comment by Agrrox** [comment]
Surprisingly the bundle has 1.14 as a minimum so there is something wrong.

**Reply by frenzibyte** [reply]
While I doubt this can be solved, if at all, but can you please show where the 10.14 limit comes from? I don't see it anywhere, the only limit I can see is a 10.7.0 limit in the bundle itself.

**Reply by Agrrox** [reply]
I have checked `osu!.app/Contents/Info.plist` in [2022.205.0](https://github.com/ppy/osu/releases/tag/2022.205.0) and [2022.405.0](https://github.com/ppy/osu/releases/tag/2022.405.0). They are identical, both having the minimum version of the operating system required for the app to run in macOS way lower than what I have:

```
<key>LSMinimumSystemVersion</key>
<string>10.7.0</string>	
```

There is probably something else deeper limiting this than just a property list. It could be set by the frameworks used and as you mentioned it's something you can't do anything about. I will try to find what's causing this, and I find something I will report it to the discussion.

**Reply by frenzibyte** [reply]
I'm fully aware about that limitation, and if you can't find the 10.14 minimum limit anywhere then it's safe to say it's beyond our scope.

**Reply by turbedi** [reply]
Probably from here (.NET 6 Release branch from the runtime): https://github.com/dotnet/runtime/blob/2eb4efb129b2c0ed79069ac5dc9a46a64675b85e/eng/native/configurecompiler.cmake#L423
It looks like .NET 6 still supports 10.14, although the documented minimum version is 10.15

**Reply by frenzibyte** [reply]
Brought to notice in https://github.com/dotnet/runtime/issues/48721.
